statsd/statsd.go

```diff
@@ -71,7 +83,14 @@ var (
 // A Client is a handle for sending udp messages to dogstatsd.  It is safe to
 // use one Client from multiple goroutines simultaneously.
 type Client struct {
+	// Address to send metrics to, needed to allow reconnection on error
+	addr net.Addr
+	// Established connection object, or nil if Connect needed
 	conn net.Conn
+	// function pointer to use to write data to the connection
+	write func([]byte) error
```

Comment:
🍰  If we export `Write`, `Client` would implement the `Writer` interface, which might be desireable (I'm not sure just a thought). Although we might also want precisely the opposite to prevent the users from writing directly to the socket 🤔